pywt/_extensions/_cwt.pyx

```diff
@@ -0,0 +1,108 @@
+cimport common, c_cwt
+from common cimport index_t, MODE
+from ._pywt cimport _check_dtype
+
+cimport numpy as np
+import numpy as np
+
+
+
+
+cpdef cwt_psi_single(data_t[::1] data, Wavelet wavelet, size_t output_len):
+    cdef data_t[::1] psi
+
+    if output_len < 1:
+        raise RuntimeError("Invalid output length.")
+
+    #if data_t is np.float64_t:
+        # TODO: Don't think these have to be 0-initialized
+        # TODO: Check other methods of allocating (e.g. Cython/CPython arrays)
+    if data_t is np.float64_t:
+        psi = np.zeros(output_len, np.float64)
+        #data = np.zeros(output_len, np.float64)
+        #data = np.linspace(wavelet.w.lower_bound, wavelet.w.upper_bound, output_len,dtype=np.float64)
+        if wavelet.short_family_name == "gaus":
+            c_cwt.double_gaus(&data[0], &psi[0], data.size, wavelet.number)
+        elif wavelet.short_family_name == "mexh":
+            c_cwt.double_mexh(&data[0], &psi[0], data.size)
+        elif wavelet.short_family_name == "morl":
+            c_cwt.double_morl(&data[0], &psi[0], data.size)
+    elif data_t is np.float32_t:
+        psi = np.zeros(output_len, np.float32)
+    #    data = np.linspace(self.w.lower_bound, self.w.upper_bound, output_len)
+    #    data = np.array(data, dtype=np.float32)
+        if wavelet.short_family_name == "gaus":
+            c_cwt.float_gaus(&data[0], &psi[0], data.size, wavelet.number)
+        elif wavelet.short_family_name == "mexh":
+            c_cwt.float_mexh(&data[0], &psi[0], data.size)
+        elif wavelet.short_family_name == "morl":
+            c_cwt.float_morl(&data[0], &psi[0], data.size)
+
+    return psi
+
+
+  
+    
+
+
+cpdef cwt_conv(data_t[::1] data, data_t[::1] in_filter, size_t output_len):
+    cdef size_t N, F, O
```

Comment:
When building I got:
`warning: _cwt.pyx:71:18: Index should be typed for more efficient access`
It seems the index, `i`, used in `cwt_conv` needs to be included in the list of `size_t` cdefs